src/core/count_min_sketch.cc

```diff
@@ -0,0 +1,123 @@
+// Copyright 2025, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#include "core/count_min_sketch.h"
+
+#include <absl/time/clock.h>
+#include <xxhash.h>
+
+#include <cmath>
+#include <functional>
+#include <iostream>
+
+namespace {
+
+constexpr auto MAX = std::numeric_limits<dfly::CountMinSketch::SizeT>::max();
+
+uint64_t GetCurrentMS() {
+  return absl::GetCurrentTimeNanos() / 1000 / 1000;
+}
+
+uint64_t ExponentialDecay(uint64_t value, int64_t time_delta) {
+  // Value halves every 5000 ms: ln(2) / 5000
+  static constexpr double EXP_DECAY_CONST = 0.000138629;
+
+  return value * std::exp(-time_delta * EXP_DECAY_CONST);
+}
+
+uint64_t LinearDecay(uint64_t value, int64_t time_delta) {
+  // Value decrements by one every 1000 ms
+  static constexpr double LIN_DECAY_CONST = 0.001;
+
+  const double decay = time_delta * LIN_DECAY_CONST;
+  return value - std::min(static_cast<double>(value), decay);
+}
+
+using DecayFn = std::function<uint64_t(uint64_t, int64_t)>;
```

Comment:
Function pointer is enough, though std function of one shouldn't incurr any overhead